pkg/certificates/certificates_test.go

```diff
@@ -616,16 +616,12 @@ func TestCertificateSigner_Sign_E2E(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, serverIdent)
 
-	listener, err := net.Listen("tcp", "127.0.0.1:0")
-	require.NoError(t, err)
-	require.NotNil(t, listener)
-
-	serverConfig := server.Config{Address: listener.Addr().String()}
-	opts, err := tlsopts.NewOptions(serverIdent, serverConfig.Config)
+	sc := server.Config{Address: "127.0.0.1:0", PrivateAddress: "127.0.0.1:7778"}
```

Comment:
Tests shouldn't use designated ports, they developers have all kinds of stuff running on their local system and it's likely to conflict with something. Use `127.0.0.1:0`.